From 56eacff9fa3a84b19ac9b8bb7072d9b7d96755e7 Mon Sep 17 00:00:00 2001
From: anthraxx <levente@leventepolyak.net>
Date: Sat, 13 Feb 2021 04:56:30 +0100
Subject: [PATCH] override main module version which we can be filled with the
 correct version

Go BuildInfo only works if we build from a module, however we simply
want to build in tree. Therefor override the main module version with
something that we can dynamically replace.
---
 caddy.go | 1 +
 1 file changed, 1 insertion(+)

diff --git a/caddy.go b/caddy.go
index 70135ffb..04d95716 100644
--- a/caddy.go
+++ b/caddy.go
@@ -679,6 +679,7 @@ func goModule(mod *debug.Module) *debug.Module {
 	mod.Version = "unknown"
 	bi, ok := debug.ReadBuildInfo()
 	if ok {
+		bi.Main.Version = "unknown"
 		mod.Path = bi.Main.Path
 		// The recommended way to build Caddy involves
 		// creating a separate main module, which
-- 
2.30.0

